Libs/Visualization/VTK/Widgets/Testing/Cpp/ctkVTKDiscretizableColorTransferWidgetTest1.cpp

```diff
@@ -59,9 +61,18 @@ int ctkVTKDiscretizableColorTransferWidgetTest1(int argc, char * argv[])
   ctf0->AddRGBPoint(0.0, 1.0, 0.0, 1.0);
   ctf0->AddRGBPoint(255.0, 0.0, 1.0, 0.0);
 
+  /// Create an image to test the widget histogram
+  vtkNew<vtkRTAnalyticSource> imageSource;
+  imageSource->SetWholeExtent(0, 9,
+                              0, 9,
+                              0, 9);
+
   /// Discretizable transfer function widget
   ctkVTKDiscretizableColorTransferWidget mWidget;
-  mWidget.setColorTransferFunction(dctf0.Get());
+  mWidget.copyColorTransferFunction(dctf0.Get());
+  mWidget.setHistogramConnection(imageSource->GetOutputPort());
```

Comment:
Could you also test "SetHistogramInputData" ? (e.g make sure it doesn't crash if setting to 0, ...)

Reply:
"SetHistogramInputData" doesn't exist, it's not a real VTK filter so it doesn't exist by default and I didn't develop it.

Reply:
Was thinking about this one https://github.com/tcoulange/CTK/blob/49d1956c645eb19d9a5317601c8d150ded450190/Libs/Visualization/VTK/Core/vtkScalarsToColorsHistogramChart.h#L56-L57

but since it is indrectly called by `ctkVTKDiscretizableColorTransferWidget::updateHistogram`, we are all set.

Reply:
Unfortunately I didn't manage to use smartpointers, and I'm not sure calling "onRowsAboutToBeRemoved" from the destructor is much better... I replaced QComboBox::clear() by this->clear(), hopefully this is sufficient.